tools/ob_admin/backup_tool/ob_admin_backup_validation_task.cpp

```diff
@@ -0,0 +1,4508 @@
+                     != macro_block_id_mappings_meta_index_list.at(k)->turn_id_) {
+            ret = OB_ERR_UNEXPECTED;
+            STORAGE_LOG(WARN, "turn id not match", K(ret),
+                        K(sstable_meta_index_list.at(i)->turn_id_),
+                        K(macro_block_id_mappings_meta_index_list.at(k)->turn_id_));
+          } else if (sstable_meta_index_list.at(i)->retry_id_
+                     != macro_block_id_mappings_meta_index_list.at(k)->retry_id_) {
+            ret = OB_ERR_UNEXPECTED;
+            STORAGE_LOG(WARN, "retry id not match", K(ret),
+                        K(sstable_meta_index_list.at(i)->retry_id_),
+                        K(macro_block_id_mappings_meta_index_list.at(k)->retry_id_));
+          } else {
+            i++;
+          }
+        }
+        if (OB_SUCC(ret)) {
+          j++;
+          k++;
+        }
+      }
+    }
+    if (OB_FAIL(ret) || i != sstable_meta_index_list.count() || j != tablet_meta_index_list.count()
+        || k != macro_block_id_mappings_meta_index_list.count()) {
+      ret = OB_ERR_UNEXPECTED;
+      STORAGE_LOG(WARN, "process not finish", K(ret), K(i), K(j), K(k));
+    } else {
+      STORAGE_LOG(INFO, "check tablet meta index finish", K(ret), K(i), K(j), K(k));
+    }
+  }
+  return ret;
+}
+int ObAdminBackupSetMetaValidationTask::check_tenant_tablet_meta_index()
+{
+  // cross check between tenant minor/major meta index from initial to end
+  int ret = OB_SUCCESS;
+  int64_t backup_set_id = -1;
+  ObAdminBackupSetValidationAttr *backup_set_attr = nullptr;
+  common::ObArray<std::pair<share::ObLSID, common::ObTabletID>> missing_tablet;
+  if (IS_NOT_INIT) {
+    ret = OB_NOT_INIT;
+    STORAGE_LOG(WARN, "not init", K(ret));
+  } else if (OB_FAIL(ctx_->processing_backup_set_id_array_.at(task_id_, backup_set_id))) {
+    STORAGE_LOG(WARN, "failed to get backup set id", K(ret), K(task_id_));
+  } else if (OB_FAIL(ctx_->get_backup_set_attr(backup_set_id, backup_set_attr))) {
+    STORAGE_LOG(WARN, "failed to get backup set store", K(ret), K(backup_set_id));
+  } else if (OB_ISNULL(backup_set_attr)) {
+    ret = OB_ERR_UNEXPECTED;
+    STORAGE_LOG(WARN, "backup data store is null", K(ret));
+  } else if (backup_set_attr->minor_tablet_id_.count()
+             != backup_set_attr->minor_tablet_map_.size()) {
+    ret = OB_ERR_UNEXPECTED;
+    STORAGE_LOG(WARN, "minor tablet id size not equal minor tablet map size", K(ret),
+                K(backup_set_attr->minor_tablet_id_.count()),
+                K(backup_set_attr->minor_tablet_map_.size()));
+  } else if (backup_set_attr->major_tablet_id_.count()
+             != backup_set_attr->major_tablet_map_.size()) {
+    ret = OB_ERR_UNEXPECTED;
+    STORAGE_LOG(WARN, "major tablet id size not equal major tablet map size", K(ret),
+                K(backup_set_attr->major_tablet_id_.count()),
+                K(backup_set_attr->major_tablet_map_.size()));
+  } else if (backup_set_attr->minor_tablet_id_.count()
```

Comment:
is this correct?

Reply:
this is absolute, since i only add tablet at consistent scn.
https://github.com/jscblack/oceanbase/blob/88e20bbc6c36d0c5e7045e3dc164f7f9dd43ad2d/tools/ob_admin/backup_tool/ob_admin_backup_validation_task.cpp#L2949